### PR TITLE
Use crypto.diffieHellman to derive secret if available

### PR DESCRIPTION
**Pull request by mattcollier**

### CODE REVIEW

**Review by aljones15 — APPROVED** [review]
I don't totally understand what is happening here, but if we are not using nacl all the time and instead using a node crypto method that seems fine. I am assuming the es2020 syntax has been tested when used in other projects? What version of nodeJs does this require?

this spec is involved with this?
https://tools.ietf.org/html/rfc3526

**Comment by mattcollier** [comment]
@aljones15 Turns out that converting an Ed25519 key into a X25519 key is computationally expensive and slow.  It also happens that Node.js recently landed a native implementation we can use for this.  So, we use native code when available, otherwise, fall back to using nacl.

There's a comment that states that the `diffieHelman` API (not to be confused with the `DiffieHellman` class was added in 13.x).  If the API is present, it will be used, if not nacl will be used.

Dynamic imports, the ES2020 feature in question, are supported in both node 13 and 14 which means the existing check for the `crypto.diffieHelman` API should be adequate to also ensure that dynamic imports are available.

File: EncryptTransformer.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*!
- * Copyright (c) 2019 Digital Bazaar, Inc. All rights reserved.
+ * Copyright (c) 2019-2020 Digital Bazaar, Inc. All rights reserved.
  */
 'use strict';
```
Comment (dlongley) [inline EncryptTransformer.js]:
```suggestion
```

File: DecryptTransformer.js [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /*!
- * Copyright (c) 2019 Digital Bazaar, Inc. All rights reserved.
+ * Copyright (c) 2019-2020 Digital Bazaar, Inc. All rights reserved.
  */
 'use strict';
```
Comment (dlongley) [inline DecryptTransformer.js]:
```suggestion
```

File: Cipher.js [hunk 3]
```diff
@@ -1,5 +1,5 @@
 /*!
- * Copyright (c) 2019 Digital Bazaar, Inc. All rights reserved.
+ * Copyright (c) 2019-2020 Digital Bazaar, Inc. All rights reserved.
  */
 'use strict';
```
Comment (dlongley) [inline Cipher.js]:
```suggestion
```

**Review by dlongley — CHANGES_REQUESTED** [review]
I think we shouldn't add `env` to this project -- at least not in the way it's being used here. I'll take on switching this over to use the existing pattern.

**Comment by dlongley** [comment]
In fact, I get this error when trying to run the karma tests locally:

```
ERROR in ../algorithms/diffieHellman.js
Module not found: Error: Can't resolve 'crypto' in '../minimal-cipher/algorithms'
 @ ../algorithms/diffieHellman.js 20:25-41 49:25-41 72:31-47
```

I'm not sure why they are passing with github actions. I'm going to switch things over and see if they pass locally.

**Review by dlongley — APPROVED** [review]
Applied changes to remove `env.js`. Solves the test-karma issue as well. LGTM now. Thanks, @mattcollier!

File: .eslintrc.js [hunk 4]
```diff
@@ -12,5 +12,8 @@ module.exports = {
     TextDecoder: true,
     TextEncoder: true,
     Uint8Array: true
+  },
+  parserOptions: {
+    ecmaVersion: 2020
```
Comment (dlongley) [inline .eslintrc.js]: If this was only for the dynamic import, we can remove now.

**Comment by dlongley** [comment]
@mattcollier - If you're ok with the changes I made above then, IMO, this is ready for merge and release.

**Pull request merged**